src/main/java/com/redhat/rhjmc/containerjfr/MainModule.java

```diff
@@ -85,7 +87,11 @@ static Logger provideLogger() {
     @Provides
     @Singleton
     public static Gson provideGson() {
```

Comment:
Could you inject a `Logger` instance here and pass it to `new GsonJmxServiceUrlAdapter(logger)`?

Reply:
could I just inject the logger into the `GsonJmxServiceUrlAdapter`? I had trouble figuring out how to inject into the static method provideGson()

Reply:
All you do is add the Logger to the parameter list of this method. Since it's part of a Dagger module and has the `@Provides` annotation, Dagger will see that and inject the logger instance into this method.

Reply:
Just scroll down a few lines within this file for an example.

```
    @Provides
    static Path provideSavedRecordingsPath(Logger logger, Environment env) {
      ...
    }
```

This is a cleaned-up version of the next method down, which has two injected dependencies.

Reply:
If you want to inject the Logger directly into the TypeAdapter then that's also an option, but then rather than directly `new`-ing it here, you will have to add an `@Provides` or `@Binds` method to some module to allow the adapter to have dependencies injected, and to allow the adapter itself to be injected into other places.